underscore.js

```diff
@@ -1340,8 +1347,16 @@
 
   // Shortcut function for checking if an object has a given property directly
   // on itself (in other words, not on a prototype).
-  _.has = function(obj, key) {
-    return obj != null && hasOwnProperty.call(obj, key);
+  _.has = function(object, path) {
+    if (!_.isArray(path)) path = [path];
```

Comment:
I had a similar concern. Especially for `_.has`. Do you have any ideas of ways we could validate that choice, or does it seem like an obviously worthwhile performance tweak? I must admit, the performance characteristics of JavaScript, and especially performance _tests_ are often anti-intuitive to me.